src/web/sitemap.rs

```diff
@@ -127,13 +159,15 @@ mod tests {
         wrapper(|env| {
             let web = env.frontend();
             assert_success("/sitemap.xml", web)?;
+            assert_success("/-/sitemap/s/sitemap.xml", web)?;
 
             env.fake_release().name("some_random_crate").create()?;
             env.fake_release()
                 .name("some_random_crate_that_failed")
                 .build_result_successful(false)
                 .create()?;
-            assert_success("/sitemap.xml", web)
+            assert_success("/sitemap.xml", web)?;
+            assert_success("/-/sitemap/s/sitemap.xml", web)
```

Comment:
Can you test all 26 letters exhaustively, and add a test that other pages give a 404?

Reply:
I extended the tests, 

also I let the sub-sitemap handler raise 404 in some more cases.